### ISSUE TITLE
Strict parameter type inspection

### ISSUE THREAD

**Issue by donquixote** [kalessil/phpinspectionsea#1876]
In PhpStorm, inspections for parameter types are quite weak:
If a variable inferred as `I|J` is passed as parameter with type `I`, PhpStorm won't complain:
https://youtrack.jetbrains.com/issue/WI-71173/Parameter-type-inspection-on-nullable-types-or-union-type-does-not-work

> That's why it was decided to show inspection warnings only for the cases when the problem definitely will occur. But if you still want such cases to be highlighted, you can use integration with PhpStan/Psalm and enable related inspections

```php
interface I {}
interface J {}
interface K {}

abstract class Test1 {
  function f(I $v_i, ?I $v_in, I|J $v_ij) {
    $this->f_j($v_i);  // "Expected parameter of type 'J', 'I' provided"
    $this->f_j($v_in);  // "Expected parameter of type 'J', 'I|null' provided"
    $this->f_jn($v_i);  // "Expected parameter of type 'J|null', 'I' provided"
    $this->f_jn($v_in);  // No inspection warning.
    $this->f_j($v_ij);  // No inspection warning.
    $this->f_jk($v_ij);  // No inspection warning.
  }
  abstract function f_j(J $v);
  abstract function f_jn(?J $v);
  abstract function f_jk(J|K $v);
}
```

Given that PhpStorm won't implement a more strict inspection, could Inspections EA provide an alternative?
